File: backend/PROHUB.API/PROHUB.API.Server/Entities/ContextDoc.cs
namespace PROHUB.API.Server.Entities;

public class ContextDoc
{
    public Guid Id { get; set; }
    public Guid ProjectId { get; set; }
    public string Content { get; set; } = "";
    public DateTime UpdatedAtUtc { get; set; }
    public string? UpdatedBy { get; set; }

    public Project? Project { get; set; }
    public ICollection<ContextDocVersion> Versions { get; set; } = [];
}

File: backend/PROHUB.API/frontend/src/api/contextDoc.ts
import { api } from './client';
import type { ContextDoc, ContextDocVersion } from './types';

export const contextDocApi = {
  get: (projectId: string) => api.get<ContextDoc>(`/projects/${projectId}/context-doc`),
  save: (projectId: string, body: { content: string; updatedBy?: string }) =>
    api.put<ContextDoc>(`/projects/${projectId}/context-doc`, body),
  versions: (projectId: string) =>
    api.get<ContextDocVersion[]>(`/projects/${projectId}/context-doc/versions`),
};
